packages/vm/src/eei/vmState.ts

```diff
@@ -286,6 +325,7 @@ export class VmState implements EVMStateAccess {
       }
     }
     this._touched.clear()
+    this._touchedStack = {}
```

Comment:
The `touchedStack` is not empty for some reason here, which makes me not super confident that this is all correct. After running `npm run test:blockchain -- --fork=Istanbul --test=RevertPrecompiledTouch_d0g0v0_Istanbul` there are still fields at the `0` and `2` keys.